docs/reference/remap/begins_with.cue

```diff
@@ -0,0 +1,55 @@
+package metadata
+
+remap: functions: begins_with: {
+	arguments: [
+		{
+			name:        "value"
+			description: "The text to search."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name:        "substring"
+			description: "The substring to search for in `value`."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name:        "case_sensitive"
+			description: "Should the match be case sensitive?"
+			required:    false
+			type: ["boolean"]
+			default: true
+		},
+	]
+	return: ["boolean"]
+	category: "text"
+	description: #"""
+			Determines if a given string begins with a given `substring`.
+			The search can be optionally case insensitive.
+		"""#
+	examples: [
+		{
+			title: "Success"
+			input: {
+				message: #"The Needle In The Haystack"#
+			}
+			source: #"""
+				.contains = begins_with(.message, "the needle", case_sensitive = false)
+				"""#
+			output: {
+				contains: true
+			}
+		},
+		{
+			title: "Error"
+			input: {
+				message: "A string with 42"
+			}
+			source: ".contains = begins_with(.message, 42)"
+			output: {
+				error: remap.errors.ArgumentError
+			}
```

Comment:
It's unclear to me what this represents.

Does `error: ` mean it would return an object with an `error` key, similar to how the `Success` example shows the final JSON payload?

Also, what does `ArgumentError` represent? Is this something we still need to implement in remap-lang?

Reply:
If you look in remap.cue there is a map of these to descriptions about why the errors occurred:
```
		ArgumentError: {
			description: "Raised when the provided input is not a supported type."
		}
```